agent/structs/config_entry_gateways.go

```diff
@@ -928,6 +929,75 @@ func (a *APIGatewayTLSConfiguration) IsEmpty() bool {
 	return len(a.Certificates) == 0 && len(a.MaxVersion) == 0 && len(a.MinVersion) == 0 && len(a.CipherSuites) == 0
 }
 
+type ServiceKey string
+
+func NewServiceKey(name string, entMeta *acl.EnterpriseMeta) ServiceKey {
```

Comment:
Why not just use `ServiceName` here? It should already contain all of this info, has a `String()` implementation, and can be parsed via `ServiceNameFromString`.